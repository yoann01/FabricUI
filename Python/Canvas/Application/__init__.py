__all__ = [
	'FabricApplicationStates',
	'FabricLog'
]
