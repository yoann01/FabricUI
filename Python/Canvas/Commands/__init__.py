__all__ = [
	'CommandHelpers',
	'CommandManager',
  'CommandRegistry',
  'CommandManagerCallback'
]
