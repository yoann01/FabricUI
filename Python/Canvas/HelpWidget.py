
from PySide import QtCore, QtGui

class HelpWidget(QtGui.QWidget):

    """HelpWidget

    HelpWidget dispalys within a QTextEdit the application usage.
    The usage description is defined in a external file (eg .txt).

    Arguments:
        usagetFilePath (String): The path of the usage file.
        parent (QWidget): The menu widget parent, can be None.
        width (int): The widget width, 800 by default.
        height (int): The widget height, 500 by default.

    """

    def __init__(self, usagetFilePath, parent = None, width = 800, height = 500):
        super(HelpWidget, self).__init__(parent)
        
        self.txtEdit = QtGui.QTextEdit(self)
        self.__readfile(usagetFilePath)
        self.txtEdit.setWindowFlags(QtCore.Qt.WindowStaysOnTopHint)
        self.txtEdit.setReadOnly( True )
        closeButton = QtGui.QPushButton("close", self)
     
        vbox = QtGui.QVBoxLayout()
        vbox.addWidget(self.txtEdit)
        vbox.addWidget(closeButton)
        self.setLayout(vbox)  

        self.setSizePolicy(QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Expanding);
        self.resize(width, height)

        closeButton.clicked.connect(self.close)

    def __readfile(self, path):
        file_ = QtCore.QFile(path);
        
        if file_ is not None:
            self.txtEdit.clear();
            if file_.open(QtCore.QIODevice.ReadOnly | QtCore.QIODevice.Text):
                stream = QtCore.QTextStream(file_)
                while not stream.atEnd():
                    line = stream.readLine()
                    self.txtEdit.setText(self.txtEdit.toPlainText()+ line + str("\n"))  
            file_.close()
