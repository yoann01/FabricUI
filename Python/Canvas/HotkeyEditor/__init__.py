__all__ = [
	'HotkeyActions',
	'HotkeyEditorDialog',
	'HotkeyTableWidget',
	'HotkeyCommands',
	'HotkeyStyledItemDelegate',
	'HotkeyTableWidgetItems'
]