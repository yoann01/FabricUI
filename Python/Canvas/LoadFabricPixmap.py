#
# Copyright (c) 2010-2017 Fabric Software Inc. All rights reserved.
#

import os
from PySide import QtGui

def LoadFabricPixmap(basename):
    """Loads the contents of the pixmap below
    
    Args:
        basename (str): Stylesheet base name.
    
    Returns:
        pixmap
    
    """

    fabricDir = os.environ.get('FABRIC_DIR')
    if fabricDir is None:
        raise IOError("LoadFabricPixmap: missing FABRIC_DIR environment variable\n")
    path = os.path.join(fabricDir, "Resources", basename)

    return QtGui.QPixmap(path)
