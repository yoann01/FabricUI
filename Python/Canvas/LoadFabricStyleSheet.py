import os
import platform


def LoadFabricStyleSheet(basename):
    """Loads the contents of the stylesheet and formats it for compatibility 
    with Fabric Engine widgets.
    
    Args:
        basename (str): Stylesheet base name.
    
    Returns:
        str: Formatted contents of the stylesheet file.
    
    """

    fabricDir = os.environ.get('FABRIC_DIR')
    if fabricDir is None:
        raise IOError("LoadFabricStyleSheet: missing FABRIC_DIR environment variable\n")
    
    qssPath = os.path.join(fabricDir, 
                           "Resources",
                           "QSS",
                           basename)

    qssData = ''

    try:
        with open(qssPath, "r") as qssFile:
            qssData = os.path.expandvars(qssFile.read())
    except EnvironmentError:
        print "LoadFabricStyleSheet: unable to open {}\n".format(qssPath)
        return qssData

    if qssData == '':
        print "LoadFabricStyleSheet: style sheet {} is empty\n".format(qssPath)
        return qssData

    if platform.system() == 'Windows':
        qssData = qssData.replace("\\", "/")

    return qssData