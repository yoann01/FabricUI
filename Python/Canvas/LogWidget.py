"""
The LogWidget receives and saves logging output from the ScriptEditor.
"""

from PySide import QtCore, QtGui
from FabricEngine.FabricUI import DFG, Actions

class AppendingTextWidget(QtGui.QTextEdit):
    def __init__(self):
        QtGui.QTextEdit.__init__(self)

        self.setReadOnly(True)
        self.prevCursorPos = 0
        
    def append(self, text, color):
        self.setTextColor(color)
        charFormat = self.currentCharFormat()
        textCursor = self.textCursor()
        textCursor.movePosition(QtGui.QTextCursor.End)
        self.prevCursorPos = textCursor.position()
        textCursor.insertText(text, charFormat)
        self.setTextCursor(textCursor)
        self.ensureCursorVisible()

    def replace(self, text, color):
        self.setTextColor(color)
        charFormat = self.currentCharFormat()
        textCursor = self.textCursor();
        textCursor.setPosition(self.prevCursorPos, QtGui.QTextCursor.KeepAnchor);
        self.moveCursor(QtGui.QTextCursor.End, QtGui.QTextCursor.KeepAnchor);
        textCursor.insertText(text, charFormat)
        self.setTextCursor(textCursor)
        self.ensureCursorVisible()

class LogWidget(AppendingTextWidget):
    def __init__(self):
        AppendingTextWidget.__init__(self)

        self.setFocusPolicy(QtCore.Qt.StrongFocus)

        self.setObjectName("ScriptEditorLogWidget")

        self.commandColor = QtGui.QColor(QtCore.Qt.white)
        self.commentColor = QtGui.QColor("#9AD6D6")
        self.exceptionColor = QtGui.QColor("#E14D59")

        self.copyAction = CopyLogAction(self)
        self.copyAvailable.connect(self.copyAction.setEnabled)

    def clearAction(self, desc):
        return ClearLogAction(self, desc)

    def appendCommand(self, text, replace = False):
        if replace is True:
            self.replace(text, self.commandColor)
        else:
            self.append(text, self.commandColor)

    def appendComment(self, text, replace = False):
        if replace is True:
            self.replace(text, self.commentColor)
        else:
            self.append(text, self.commentColor)

    def appendException(self, text, replace = False):
        if replace is True:
            self.replace(text, self.exceptionColor)
        else:
            self.append(text, self.exceptionColor)

    def contextMenuEvent(self, event):
        menu = QtGui.QMenu(self)
        menu.addAction(self.copyAction)
        menu.addSeparator()
        menu.addAction(self.clearAction("Clear"))
        menu.exec_(self.mapToGlobal(event.pos()))

class BaseLogWidgetAction(Actions.BaseAction):
 
    def __init__(self,
        logWidget, 
        name, 
        text, 
        shortcut = QtGui.QKeySequence(), 
        context = QtCore.Qt.WidgetShortcut):

        self.logWidget = logWidget

        super(BaseLogWidgetAction, self).__init__(
            logWidget, 
            name, 
            text, 
            shortcut, 
            context)
        self.logWidget.addAction(self)

class CopyLogAction(BaseLogWidgetAction):
 
    def __init__(self, logWidget):

        super(CopyLogAction, self).__init__(
            logWidget, 
            "LogWidget.CopyLogAction", 
            "Copy", 
            QtGui.QKeySequence.Copy)
        
        self.setEnabled(self.logWidget.textCursor().hasSelection())

    def onTriggered(self):
        self.logWidget.copy()

class ClearLogAction(BaseLogWidgetAction):
 
    def __init__(self, logWidget, desc):

        super(ClearLogAction, self).__init__(
            logWidget, 
            "LogWidget.clearAction", 
            desc)

    def onTriggered(self):
        self.logWidget.clear()
