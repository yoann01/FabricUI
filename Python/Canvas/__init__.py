__all__ = [
    'Utils',
    'BindingWrapper',
    'HotkeyEditor',
    'CanvasWindow',
    'FabricParser',
    'LogWidget',
    'PythonHighlighter',
    'RTValEncoderDecoder',
    'ScriptEditor',
    'UICmdHandler',
]
