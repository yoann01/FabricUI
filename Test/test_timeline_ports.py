import os, pytest
import fabric_test_ui
from PySide import QtCore
from FabricEngine import Core
from FabricEngine.FabricUI import Application
from FabricEngine.Canvas.CanvasWindow import CanvasWindow

# [andrew 20160330] FE-6364
pytestmark = pytest.mark.skipif(
    fabric_test_ui.is_missing_display(),
    reason="missing display",
    )

@pytest.yield_fixture(scope="module")
def canvas_app():
    app = fabric_test_ui.create_canvas_app()
    yield app
    fabric_test_ui.cleanup_canvas_app(app)

@pytest.yield_fixture(scope="module")
def canvas_win(canvas_app):
    win = fabric_test_ui.create_canvas_win(
      canvas_app,
      report_line_mapper=lambda line: line + "\n",
      )
    yield win
    fabric_test_ui.cleanup_canvas_win(win)

# Returns the output of the test
def main(canvas) :
  from PySide import QtCore
  from FabricEngine import Core
  binding = canvas.dfgWidget.getDFGController().getBinding()
  ex = binding.getExec()

  names = [
    "timeline",
    "timelineStart",
    "timelineEnd",
    "timelineFramerate"
  ]
  for name in names :
    ex.addExecPort( name, canvas.client.DFG.PortTypes.In, "Float32" )

  funcName = "reportPorts"
  ex.addInstWithNewFunc( funcName )
  func = ex.getSubExec( funcName )

  for name in names :
    func.addExecPort( name, canvas.client.DFG.PortTypes.In, "Float32" )
    ex.connectTo( name, funcName + '.' + name )

  mainCode = "\n".join( [ " report( '" + name + " = ' + " + name + " );" for name in names ] )
  func.setCode("""
  dfgEntry
  {
  """ + mainCode + """
  report("");
  }
  """)

  ex.connectTo( funcName + '.exec', 'exec' )

  #print("TEST_START");
  canvas.test_output = ""


  # Default values
  QtCore.QCoreApplication.processEvents()

  # Time Range
  canvas.timeLine.setTimeRange( 11, 314 )
  QtCore.QCoreApplication.processEvents()

  # FrameRate
  canvas.timeLine.setFrameRate( 25 )
  QtCore.QCoreApplication.processEvents()

  # Back to an empty graph
  canvas.show()
  canvas.onNewGraph(True)

  #print("TEST_END");
  return canvas.test_output

def test_timeline_ports(canvas_win):

  import os

  refOutputPath = os.environ['FABRIC_SCENE_GRAPH_DIR'] + '/Native/FabricUI/Test/test_timeline_ports.out'
  refOutput = open( refOutputPath, 'r' ).read()

  refBaseName, refOutputExt = os.path.splitext( refOutputPath )
  if( os.path.exists( refBaseName + os.extsep + os.environ['FABRIC_BUILD_OS'] + os.extsep + "skip" ) ) :
    import pytest
    pytest.skip("Skip")

  # TODO: some of the output from the DFGController is not caught
  testOutput = main(canvas_win)

  if testOutput != refOutput :
    import difflib
    diff = difflib.unified_diff( refOutput.splitlines(1), testOutput.splitlines(1) )
    raise Exception( ''.join( diff ) )
