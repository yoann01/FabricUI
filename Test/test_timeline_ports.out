timeline = +1.0
timelineStart = +1.0
timelineEnd = +250.0
timelineFramerate = +60.0

timeline = +11.0
timelineStart = +1.0
timelineEnd = +250.0
timelineFramerate = +60.0

timeline = +11.0
timelineStart = +11.0
timelineEnd = +314.0
timelineFramerate = +60.0

timeline = +11.0
timelineStart = +11.0
timelineEnd = +314.0
timelineFramerate = +25.0

